Wrong return type in wrapper for a class taking an untranslatable argument

### Describe the bug
When a class argument cannot be wrapped the function `_get_untranslatable_function` is used however this function gives the wrong result type for a class init function.

### To Reproduce
Provide code to reproduce the behavior:
See `tests/warnings/wrapper/HIGH_ORDER_FUNCTIONS_IN_CLASS_FUNCS.py`

### Error details
Provide the generated code, or the error message:
```
/home/emily/Code/pyccel/tests/warnings/wrapper/__pyccel__/HIGH_ORDER_FUNCTIONS_IN_CLASS_FUNCS_wrapper.c:44:12: warning: returning ‘int’ from a function with return type ‘PyObject *’ {aka ‘struct _object *’} makes pointer from integer without a cast [-Wint-conversion]
   44 |     return -1;
      |            ^

  warnings.warn(UserWarning(err))
```

### Language
C

### Additional context
This is now an error on the MacOS CI

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -51,7 +51,7 @@ All notable changes to this project will be documented in this file.
 -   #1927 : Improve error Message for missing target language compiler in Pyccel
 -   #1933 : Improve code printing speed.
 -   #1930 : Preserve ordering of import targets.
--   #1951 : Fix return type for class taking unwrappable argument
+-   #1951 : Fix return type for class whose argument cannot be wrapped.
 
 ### Changed
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -51,6 +51,7 @@ All notable changes to this project will be documented in this file.
 -   #1927 : Improve error Message for missing target language compiler in Pyccel
 -   #1933 : Improve code printing speed.
 -   #1930 : Preserve ordering of import targets.
+-   #1951 : Fix return type for class taking unwrappable argument
 
 ### Changed
 

--- a/pyccel/codegen/wrapper/c_to_python_wrapper.py
+++ b/pyccel/codegen/wrapper/c_to_python_wrapper.py
@@ -443,7 +443,10 @@ class CToPythonWrapper(Wrapper):
             The new function which raises the error.
         """
         func_args = [FunctionDefArgument(self.get_new_PyObject(n)) for n in ("self", "args", "kwargs")]
-        func_results = [FunctionDefResult(self.get_new_PyObject("result", is_temp=True))]
+        if self._error_exit_code is Nil():
+            func_results = [FunctionDefResult(self.get_new_PyObject("result", is_temp=True))]
+        else:
+            func_results = [FunctionDefResult(self.scope.get_temporary_variable(self._error_exit_code.class_type, "result"))]
         function = PyFunctionDef(name = name, arguments = func_args, results = func_results,
                 body = [FunctionCall(PyErr_SetString, [PyNotImplementedError,
                                         LiteralString(error_msg)]),

